File: HackerRank/DataStructures/LinkedLists/insert_at_head.cpp
SinglyLinkedListNode* insertNodeAtHead(SinglyLinkedListNode* llist, int data) {
    SinglyLinkedListNode* newnode = new SinglyLinkedListNode(data);
    
    if(llist == nullptr){
        llist = newnode;
        return llist;
    }
    
    newnode -> next = llist;
    return newnode;
}

File: HackerRank/DataStructures/LinkedLists/InsertAtHead.java
static SinglyLinkedListNode insertNodeAtHead(SinglyLinkedListNode llist, int data) {
    SinglyLinkedListNode newnode = new SinglyLinkedListNode(data);
    
    if(llist == null){
        llist = newnode;
        return llist;
    }
    
    newnode.next = llist;
    return newnode;
}